src/modules/claims/claims.service.ts

```diff
@@ -445,29 +443,30 @@ export class ClaimsService {
       expirationTimestamp: claimExpirationTimestamp,
     };
 
-    if (registrationTypes.includes(RegistrationTypes.OnChain)) {
+    if (
+      registrationTypes.includes(RegistrationTypes.OnChain) &&
+      publishOnChain
+    ) {
+      if (!subjectAgreement) {
+        throw new Error(
+          ERROR_MESSAGES.ONCHAIN_ROLE_SUBJECT_AGREEMENT_NOT_SPECIFIED
+        );
+      }
       const expiry = claimExpirationTimestamp || eternityTimestamp;
       const onChainProof = await this.createOnChainProof(
         role,
         version,
         expiry,
         sub
       );
-      if (!subjectAgreement) {
-        throw new Error(
-          ERROR_MESSAGES.ONCHAIN_ROLE_SUBJECT_AGREEMENT_NOT_SPECIFIED
-        );
-      }
       message.onChainProof = onChainProof;
-      if (publishOnChain) {
-        await this.registerOnchain({
-          token,
-          subjectAgreement,
-          onChainProof,
-          acceptedBy: this._signerService.did,
-          expirationTimestamp: expiry,
-        });
-      }
+      await this.registerOnchain({
+        token,
+        subjectAgreement,
+        onChainProof,
+        acceptedBy: this._signerService.did,
+        expirationTimestamp: expiry,
+      });
```

Comment:
it dont get it, it will always publish role on-chain

Reply:
Check is added in registerOnchain method